src/test/java/qna/domain/AnswerRepositoryTest.java

```diff
@@ -26,41 +25,48 @@ class AnswerRepositoryTest {
     @Autowired
     private UserRepository userRepository;
 
-    private User user1;
+    private User user;
     private Question question1;
     private Question question2;
 
     @BeforeEach
     void setUp() {
-        user1 = userRepository.save(new User("changer", "password", "name", "changer@back.end"));
-        question1 = questionRepository.save(new Question("title1", "contents1").writeBy(user1));
-        question2 = questionRepository.save(new Question("title2", "contents2").writeBy(user1));
+        user = userRepository.save(new User("changer", "password", "name", "changer@back.end"));
+        question1 = questionRepository.save(new Question("title1", "contents1").writeBy(user));
+        question2 = questionRepository.save(new Question("title2", "contents2").writeBy(user));
     }
 
     @Test
-    void 질문ID를_입력_받아_삭제되지_않은_답변을_조회한다() {
+    void 질문ID를_입력_받아_해당_질문ID의_삭제되지_않은_답변을_조회한다() {
         // given
-        answerRepository.save(new Answer(user1, question1, "Answers Contents1"));
-        Answer answer = answerRepository.save(new Answer(user1, question1, "Answers Contents2"));
-        answer.changeDeleted(true);
-
-        Answer diffrentQuestionIdAnswer = new Answer(user1, question2, "Answers Contents3");
-        answerRepository.save(diffrentQuestionIdAnswer);
+        answerRepository.save(new Answer(user, question1, "Answers Contents1"));
+        Answer answer = answerRepository.save(new Answer(user, question1, "Answers Contents2"));
+        answer.delete();
```

Comment:
라인은 3줄이지만, 의도하신 테스트조건을 이해하기위해 여러번 읽었습니다.
가독성 측면에서 조금 아쉬움이 있습니다~

제가 파악한 테스트 조건/검증하는 사항 정리해봤습니다.(혹시 다르다면 알려주세요😄~)

의도하신 테스트 조건:
1번째로 생선한 answer객체는 이름 없이 곧바로 저장됨
2번째로 생성한 'answer'를 삭제

검증하고 있는 것:
2개의 답변을 저장했는데, 2번째 답변을 삭제했으니 1개만 조회됨

-------------------------------------------

처음 읽는 사람 입장에서 가독성을 높일수 있을 만한 방법 생각해봤습니다.

- 2개의 answer에 local 변수를 부여해주면, 어떤것이 삭제되고 어떤 것은 저장된 이후 변화가 없는지 알기 쉬울것 같습니다.
- 테스트 메소드 이름에 집중해보면, '삭제되지_않은_답변을_조회한다' 인데요. 
삭제되지 않은 특정한 답변이 조회되는지 확인하는 테스트처럼 느껴집니다. 삭제 후 남은 답변의 개수 검증보다, 삭제 안되고 유지된 답변이 정확히 answer1인지 검증하는 것이 더 의도하는 바였을까요?
만약 맞다면 저라면 아래처럼 개선 할 것 같습니다.

```suggestion
        Answer answer1 = new Answer(user, question1, "Answers Contents1");
        Answer answer2 = new Answer(user, question1, "Answers Contents2")
        answerRepository.save(answer1);
        answerRepository.save(answer2);
        answer2.delete();
      
        ...
        //then
       assertThat(results.get(0).getContents()).isEqualTo("Answers Contents1");
```
저의 예시코드에서도 코드라인을 줄이거나 더 나은 local변수명 등을 고민할 필요가있겠지만,
적어도 처음 읽었을 때 헷갈릴 여지는 좀 더 줄어들 것 같습니다. 

클린코드의 세부 원칙들을 지키시는 노력도 당연히 좋지만, 
좀더 단순하게는 처음읽는 사람도 읽기 쉽고 이해하기 좋은 코드를 먼저 고려해보시면 좋을 것 같습니다.

Reply:
동의합니다! 제안해주신 대로 코드를 수정했습니다.
저희도 이전 테스트코드를 다시 읽어보니, 당시에 어떤 의도로 테스트를 했었는지 파악하기 힘들었습니다.
문제점을 짚어주시고 제안을 해주시니 너무 좋습니다. 솔라가 말씀하시는 `가독성`이 무엇인지 이해할 수 있었습니다.
앞으로 코드를 작성할 때 가독성을 높이기 위해 단순한 코드를 짜려고 노력해보겠습니다.